src/controller/backend/Endpoints/UserProfileEndpoints.cs

```diff
@@ -0,0 +1,172 @@
+            {
+                Username = user.UserName,
+                Language = user.UserProfile!.Language,
+                UseDarkMode = user.UserProfile.UseDarkMode,
+                Permissions = userPermissions.Select(s => s.Value),
+                Roles = roles,
+                RolePermissions = roleClaims
+            });
+
+        }).RequirePermission(UserClaims.ProfileViewInfo);
+
+        group.MapGet("/username", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager) =>
+        {
+            var user = await userManager.GetUserAsync(context.User);
+
+            if (user is null)
+                return Results.Unauthorized();
+
+            return Results.Ok(user.UserName!);
+
+        }).RequirePermission(UserClaims.ProfileViewProfile);
+
+        group.MapPost("/username", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager,
+            [FromBody] UsernameModel model) =>
+        {
+            var user = await userManager.GetUserAsync(context.User);
+            if (user is null)
+                return Results.Unauthorized();
+
+            if(!user.CanChangeUsername)
+                return Results.BadRequest($"Username for the '{user.UserName}' user cannot be changed");
+
+            if (string.IsNullOrWhiteSpace(model.Username))
+                return Results.BadRequest();
+
+            var result = await userManager.SetUserNameAsync(user, model.Username);
+
+            return result.Succeeded ? Results.Ok() : Results.BadRequest(result.Errors);
+
+        }).RequirePermission(UserClaims.ProfileChangeUsername);
+
+        group.MapPost("/password", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager,
+            [FromBody] PasswordModel model) =>
+        {
+            var user = userManager.GetUserAsync(context.User).Result;
+            if (user is null)
+                return Results.Unauthorized();
+
+            var result = await userManager.ChangePasswordAsync(user, model.CurrentPassword, model.NewPassword);
+
+            return result == IdentityResult.Success ? Results.Ok() : Results.BadRequest(result.Errors);
+
+        }).RequirePermission(UserClaims.ProfileChangePassword);
+
+        group.MapGet("/backup", () =>
+        {
+            // TODO: Implement backup functionality
+        }).RequireAuthorization("token");
+
+        group.MapGet("/language", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager,
+            SystemDbContext dbContext) =>
+        {
+            var user = await userManager.GetUserAsync(context.User);
+            if (user is null)
+                return Results.Unauthorized();
+            var userProfile = await dbContext.UserProfiles.FindAsync(user.UserProfileId);
+            return Results.Ok(userProfile!.Language.ToString());
+
+        }).RequirePermission(UserClaims.ProfileViewProfile);
```

Comment:
What is a ProfileViewProfile permission? Why do we need that? User only sees its own profile, admin sees all, period. This permission system is cool for data queries to have fine grained control there but how would it make sense for a user not to see their own profile or what is the idea here?